activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/api/task/TaskQueryTest.java

```diff
@@ -2661,6 +2661,41 @@ public void testLocalizeTasks() throws Exception {
     assertThat(task.getDescription()).isEqualTo("My 'en' localized description");
   }
 
+    public void testQueryCandidateOrAssignedUser() {
+        // given two new tasks assigned to the same user
+        String taskId = "test";
+        String task2Id = "test 2";
+
+        Task createdTask = taskService.newTask(taskId);
+        createdTask.setName("test task description");
+        createdTask.setPriority(0);
+        createdTask.setAssignee("hruser");
```

Comment:
The test isn't failing when reverting the Task.xml changes. I was able to reproduce by add the following to both tasks

taskService.addCandidateUser(taskId, "hruser");
taskService.addCandidateGroup(taskId, "hr");

And
removing the `setAssignee`